src/main/java/it/gov/pagopa/pu/migration/connector/debtposition/config/DebtPositionApisHolder.java

```diff
@@ -0,0 +1,65 @@
+package it.gov.pagopa.pu.migration.connector.debtposition.config;
+
+import it.gov.pagopa.pu.debtposition.client.generated.*;
+import it.gov.pagopa.pu.debtposition.generated.ApiClient;
+import it.gov.pagopa.pu.debtposition.generated.BaseApi;
+import it.gov.pagopa.pu.migration.config.rest.RestTemplateConfig;
+import jakarta.annotation.PreDestroy;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Lazy
+@Service
+public class DebtPositionApisHolder {
+    private final DebtPositionTypeOrgApi debtPositionTypeOrgApi;
+    private final DebtPositionTypeOrgEntityControllerApi debtPositionTypeOrgEntityApi;
+    private final DebtPositionTypeOrgSearchControllerApi debtPositionTypeOrgSearchControllerApi;
+
+    /** it will store the actual accessToken and mappedExternalUserId */
+    private final ThreadLocal<Pair<String, String>> authContextHolder = new ThreadLocal<>();
```

Comment:
this is not more required to be a Pair, now your are storing just the accessToken